bluepysnap/nodes.py

```diff
@@ -428,9 +428,9 @@ def ids(self, group=None, limit=None, sample=None):
             result = utils.ensure_list(group)
             self._check_ids(result)
             preserve_order = isinstance(group, collections.Sequence)
-
         if sample is not None:
-            result = np.random.choice(result, sample, replace=False)
+            if len(result) > 0:
+                result = np.random.choice(result, sample, replace=False)
             preserve_order = False
```

Comment:
Would it be correct to set `preserve_order` if `len(result) == 0?`

Reply:
The preserve_order change is really minimal because it will only result in calling the unique on an empty array and that s it. 
The factorization worth the very small overhead imao.

Reply:
Thank you for clarifying.